src/ocl/convolutionocl.cpp

```diff
@@ -140,6 +253,120 @@ static void ShrinkToFind10Results(std::vector<Solution>& found)
     found = std::move(out);
 }
 
+std::vector<Solution> VerifiedFDBSolution(const ExecutionContext& ctx,
+                                          const conv::ProblemDescription& problem,
+                                          const AnyInvokeParams& invoke_ctx,
+                                          bool force_attach_binary,
+                                          const std::vector<miopenConvSolution_t> solutions,
+                                          bool model_result)
+{
+    const auto& conv     = problem.GetConv();
+    const auto& findMode = conv.findMode;
+    auto results         = UserFindDbRecord::TryLoad(ctx.GetStream(), problem, [&]() {
+        auto ctx_copy                       = ctx;
+        ctx_copy.use_dynamic_solutions_only = findMode.IsDynamicHybrid(ctx);
+        const auto params =
+            conv::ConvFindParameters{conv.IsWinograd3x3SupportedAndFast(ctx_copy, problem)};
+
+        // test timing of solver reported by system db
+        const auto& handle = ctx_copy.GetStream();
+        AutoEnableProfiling enableProfiling{handle};
+        bool is_optimal = true;
+
+        std::vector<Solution> eval_sols;
+        auto db = MakeConvDbGetter(ctx);
+        // reverse solutions so that EvaluateInvokers registers the fastest solution last
+        auto sol_itr = solutions.rbegin();
+        auto sol_end = solutions.rend();
+        if(!model_result)
+            sol_itr = solutions.rend() - 1;
```

Comment:
Made correction to iterator, now correct solution is evaluated and compared to previous db time.